clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5574,6 +5574,12 @@ bool Sema::CheckCallingConvAttr(const ParsedAttr &Attrs, CallingConv &CC,
       A = HostTI->checkCallingConvention(CC);
     if (A == TargetInfo::CCCR_OK && CheckDevice && DeviceTI)
       A = DeviceTI->checkCallingConvention(CC);
+  } else if (LangOpts.SYCLIsDevice && (Aux && Aux->getTriple().isOSWindows()) &&
+             TI.getTriple().isAMDGPU() &&
+             getSourceManager().isInSystemHeader(Attrs.getScopeLoc())) {
```

Comment:
I don't really know what will happen if the function with a wrong calling convention will reach codegen for the device. But in general, I would not limit this fix to windows and system headers because in SYCL the users are free to do everything in the host code. To prevent errors from device code emission, we usually use DiagIfDeviceCode functions, but I'm not really a fun of them due to their bugginess .

Reply:
Okay, let me remove that part and propose it upstream, thanks